src/components/DurationDropdown/DurationDropdown.tsx

```diff
@@ -0,0 +1,61 @@
+import ToolbarDropdown from '../ToolbarDropdown/ToolbarDropdown';
+import { Durations, serverConfig } from '../../config/ServerConfig';
+import * as React from 'react';
+import { DurationInSeconds } from '../../types/Common';
+import { KialiAppState } from '../../store/Store';
+import { durationSelector } from '../../store/Selectors';
+import { ThunkDispatch } from 'redux-thunk';
+import { KialiAppAction } from '../../actions/KialiAppAction';
+import { bindActionCreators } from 'redux';
+import { UserSettingsActions } from '../../actions/UserSettingsActions';
+import { connect } from 'react-redux';
+
+type BasicDurationDropdownProps = {
```

Comment:
Thanks for adding this overdue component.  The convention is to declare ReduxProps for anything being supplied/managed by the container.  So I'd suggest removing duration and setDuration from BasicDurationDropDownProps and declaring them in ReduxProps.  Also, I think drop the 'Extended' prefix, Something like:


```
type ReduxProps = {
  duration: DurationInSeconds;
  setDuration: (duration: DurationInSeconds) => void;
}

type BasicDurationDropdownProps = {
...
}

type DurationDropdownProps = ReduxProps & BasicDurationDropdownProps & {
  durations: Durations;
}

export class DurationDropdown extends React.Component<DurationDropdownProps>

```